pkg/controller/v1alpha1/llmisvc/scheduler.go

```diff
@@ -0,0 +1,481 @@
+
+	ip := &igwapi.InferencePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-inference-pool"),
+			Namespace: llmSvc.GetNamespace(),
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+		},
+	}
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Pool != nil && llmSvc.Spec.Router.Scheduler.Pool.Spec != nil {
+		ip.Spec = *llmSvc.Spec.Router.Scheduler.Pool.Spec.DeepCopy()
+	}
+
+	log.FromContext(ctx).V(2).Info("Expected router InferencePool", "inferencepool", ip)
+
+	return ip
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerInferenceModel(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) *igwapi.InferenceModel {
+	labels := r.schedulerLabels(llmSvc)
+
+	im := &igwapi.InferenceModel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-inference-model"),
+			Namespace: llmSvc.GetNamespace(),
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+		},
+		Spec: igwapi.InferenceModelSpec{
+			ModelName: ptr.Deref(llmSvc.Spec.Model.Name, llmSvc.GetName()),
+			PoolRef: igwapi.PoolObjectReference{
+				Group: "inference.networking.x-k8s.io",
+				Kind:  "InferencePool",
+				Name:  igwapi.ObjectName(kmeta.ChildName(llmSvc.GetName(), "-inference-pool")),
+			},
+			Criticality: llmSvc.Spec.Model.Criticality,
+		},
+	}
+	if im.Spec.Criticality == nil {
+		im.Spec.Criticality = ptr.To(igwapi.Critical)
+	}
+
+	log.FromContext(ctx).V(2).Info("Expected InferenceModel", "inferencemodel", im)
+
+	return im
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerDeployment(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) *appsv1.Deployment {
+	labels := r.schedulerLabels(llmSvc)
+	d := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-kserve-router-scheduler"),
+			Namespace: llmSvc.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+			Labels: labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+			},
+		},
+	}
+
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Template != nil {
+		d.Spec.Template.Spec = *llmSvc.Spec.Router.Scheduler.Template.DeepCopy()
+		for i := range d.Spec.Template.Spec.Containers {
+			if d.Spec.Template.Spec.Containers[i].Name != "main" {
+				continue
+			}
+
+			if slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "--configText") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "-configText") ||
```

Comment:
why we use two different arg naming conventions here?

Reply:
the config can be passed and overridden by users, so if it''s overridden we don't use our default configuration